torch/csrc/Module.cpp

```diff
@@ -428,6 +428,16 @@ PyObject* THPModule_fromDLPack(PyObject* _unused, PyObject* data) {
   END_HANDLE_TH_ERRORS
 }
 
+PyObject* THPModule_isXPUAvailable_wrap(PyObject* self, PyObject* noargs) {
```

Comment:
Is there a reason some of this boilerplate / API's can't live out of tree, as part of the extension?

(1)  The python binding (e.g. `torch_xpu_extension.is_xpu_available()` )
(2) The `XPUHooks` interface + registration (the `C10_DECLARE_REGISTRY` and `C10_REGISTER_CLASS` macros can be called out of core)